src/hotspot/share/classfile/classListParser.cpp

```diff
@@ -92,6 +92,61 @@ ClassListParser::~ClassListParser() {
   _instance = NULL;
 }
 
+int ClassListParser::parse(TRAPS) {
+  int class_count = 0;
+
+  while (parse_one_line()) {
+    if (lambda_form_line()) {
+      // The current line is "@lambda-form-invoker ...". It has been recorded in LambdaFormInvokers,
+      // and will be processed later.
+      continue;
+    }
+
+    TempNewSymbol class_name_symbol = SymbolTable::new_symbol(_class_name);
+    if (_indy_items->length() > 0) {
+      // The current line is "@lambda-proxy class_name". Load the proxy class.
+      resolve_indy(THREAD, class_name_symbol);
+      class_count++;
+      continue;
+    }
+
+    Klass* klass = load_current_class(class_name_symbol, THREAD);
+    if (HAS_PENDING_EXCEPTION) {
+      if (PENDING_EXCEPTION->is_a(vmClasses::OutOfMemoryError_klass())) {
+        // If we have run out of memory, don't try to load the rest of the classes in
+        // the classlist. Throw an exception, which will terminate the dumping process.
+        return 0; // THROW
+      }
+
+      // We might have an invalid class name or an bad class. Warn about it
+      // and keep going to the next line.
+      CLEAR_PENDING_EXCEPTION;
+      log_warning(cds)("Preload Warning: Cannot find %s", _class_name);
```

Comment:
Including some information about the exception would be useful - it might not be what we think.

Reply:
This is the same as the current behavior -- all exceptions encountered during parsing of the classlist are printed as "Cannot find ...". Many of the test cases depend on this message.

I want to limit the size of this PR and stick to non-functional changes only. I plan to improve the message in do this in a separate RFE: [JDK-8263469](https://bugs.openjdk.java.net/browse/JDK-8263469) "CDS log should print specific reasons for loading failures"